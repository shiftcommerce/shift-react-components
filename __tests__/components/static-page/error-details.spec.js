// Libraries
import React from 'react'

// Components
import StaticPageErrorDetails from '../../../src/components/static-page/error-details'

const strings = [
  'Lorem ipsum',
  'Dolor sit amet',
  'Consectetur adipiscing elit'
]

const errors = {
  Endpoint: JSON.stringify(strings[0]),
  Query: JSON.stringify(strings[1]),
  'Response data': JSON.stringify(strings[2])
}

test('renders correct props and classnames', () => {
  const className = 'dummy-classname'

  // act
  const wrapper = mount(
    <StaticPageErrorDetails className={className} isProduction={false} errorDetails={errors} detailsShown />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(`Endpoint"${strings[0]}"`)
  expect(wrapper).toIncludeText(`Query"${strings[1]}"`)
  expect(wrapper).toIncludeText(`Response data"${strings[2]}"`)
  expect(wrapper.find('div').first()).toHaveClassName(className)
})

test('renders nothing if isProduction is set to true', () => {
  // act
  const wrapper = shallow(
    <StaticPageErrorDetails isProduction errorDetails={errors} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.equals(null)).toBe(true)
})
