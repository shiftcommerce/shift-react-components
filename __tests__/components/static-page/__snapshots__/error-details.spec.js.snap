// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct props and classnames 1`] = `
<StaticPageErrorDetails
  className="dummy-classname"
  detailsShown={true}
  errorDetails={
    Object {
      "Endpoint": "\\"Lorem ipsum\\"",
      "Query": "\\"Dolor sit amet\\"",
      "Response data": "\\"Consectetur adipiscing elit\\"",
    }
  }
  isProduction={false}
>
  <div
    className="dummy-classname c-static-page-error-details"
  >
    <p
      className="c-static-page-error-details__toggle"
      onClick={[Function]}
    >
      Hide details
    </p>
    <div
      className="c-static-page-error-details__list"
    >
      <p
        className="c-static-page-error-details__title"
      >
        Endpoint
      </p>
      <p
        className="c-static-page-error-details__value"
      >
        "Lorem ipsum"
      </p>
      <p
        className="c-static-page-error-details__title"
      >
        Query
      </p>
      <p
        className="c-static-page-error-details__value"
      >
        "Dolor sit amet"
      </p>
      <p
        className="c-static-page-error-details__title"
      >
        Response data
      </p>
      <p
        className="c-static-page-error-details__value"
      >
        "Consectetur adipiscing elit"
      </p>
    </div>
  </div>
</StaticPageErrorDetails>
`;

exports[`renders nothing if isProduction is set to true 1`] = `""`;
